### ISSUE TITLE
The line number display issue in the light box

### ISSUE THREAD

**Issue by jack-gaojz** [codemirror/CodeMirror#5876]
**Describe the bug**
If the codemirror in a lightbox, when the lightbox show, the codemirror linenumber display not correct:
![image](https://user-images.githubusercontent.com/40879710/57576912-49017d80-749e-11e9-8485-5b2f4118a6d9.png)
But when I scroll the scrollbar, the line number will display well:
![image](https://user-images.githubusercontent.com/40879710/57576959-fecccc00-749e-11e9-8011-1f5966ea0419.png)
If I focus on it, then type the "Enter" key, the new line number will display well, the original still has the display issue:
![image](https://user-images.githubusercontent.com/40879710/57577009-1e182900-74a0-11e9-9c60-aac0cc631264.png)

I have set the "autoRefresh" property and reference the "autoRefresh" plugin:
```
var codeEditor = $(".code-editor");
var editor = CodeMirror.fromTextArea(codeEditor.get(0), {
	mode: "text/x-csharp",
	indentWithTabs: true,
	smartIndent: true,
	lineNumbers: true,
	lineWrapping: false,
	matchBrackets: true,
	autoRefresh: false,
	extraKeys: { "Ctrl-Space": "autocomplete" }
});

editor.setSize("auto", "200px");
editor.on("change", function() {
	$("#Expression").val(editor.getValue());
});
```

Another issue is the text area connot get the codemirror value, I have to get the value via handle the "change" event to archived it.
```
editor.on("change", function() {
	$("#Expression").val(editor.getValue());
});
```
And if I want to select some codes vai the mouse in the code mirror, the display UI not make sense and the select behavor not comfortable.
![image](https://user-images.githubusercontent.com/40879710/57577069-63892600-74a1-11e9-9366-adf78c33b58e.png)

I try to use the "refresh()" to fix it, but still failed:
![image](https://user-images.githubusercontent.com/40879710/57577090-f629c500-74a1-11e9-9d07-a5d6501b883e.png)
The first two columns will not display well.
`editor.refresh();`

BTW, if the codemirror in a normal page, it works fine.

**Expected behavior**
1. The line number should display well.
2. The code select should works fine.

**Screenshots**
Above all screenshots.

**Desktop (please complete the following information):**
 - OS: Win 10 64bit
 - Browser: chrome
 - Version: 74.0.3729.108 (Official Build) (64-bit)

**Additional context**
Codemirror Version: 5.42.2

Thanks so much, the codemirror is a so amazing editor!

**Comment by berni69** [comment]
I have a similar issue, It happens when the editor is loaded in a bootstrap navtab:
![image](https://user-images.githubusercontent.com/1116133/57637461-c49d2080-75ab-11e9-89f7-d5885a5eeb2a.png)

On the main tab is working fine:

![image](https://user-images.githubusercontent.com/1116133/57637482-d383d300-75ab-11e9-9687-c05d1ebc3779.png)

I'm using the version 5.46.0

**Comment by jack-gaojz** [comment]
@berni69 I have fixed on my codes via adding the "setTimeout" to delay the codemirror load. I guess that because the codemirror calculate the html element size before they are visible. Once the html rendered, then initialize the codemirror, it will looks good.
FYI.
```
setTimeout(function () {
                var codeEditor = $(".code-editor");
                var editor = CodeMirror.fromTextArea(codeEditor.get(0), {
                    mode: "text/x-csharp",
                    indentWithTabs: true,
                    smartIndent: true,
                    lineNumbers: true,
                    lineWrapping: false,
                    matchBrackets: true,
                    autoRefresh: false,
                    extraKeys: { "Ctrl-Space": "autocomplete" }
                });

                editor.setSize("auto", "200px");
                editor.on("change", function () {
                    $("#Expression").val(editor.getValue());
                });
            }, 100);
```

**Comment by aishwaryardandi** [comment]
I have a similar issue with codemirror: 

When the codemirror is inside an accordion and if lineNumbers is set to true then the data entered overlaps with the line numbers. Also if theres any data that is already available then it doesn't appear at first, it only appears when I click inside the codemirror and again, it overlaps with the line numbers.

**Comment by jack-gaojz** [comment]
> I have a similar issue with codemirror:
> 
> When the codemirror is inside an accordion and if lineNumbers is set to true then the data entered overlaps with the line numbers. Also if theres any data that is already available then it doesn't appear at first, it only appears when I click inside the codemirror and again, it overlaps with the line numbers.

Because of the accordion content panel don't have the height value when dom ready, the codemirror should know the container height and width to calculate the dom layout. If we want to fix this issue, we should use setTimeout to delay init the codeMirror and make sure the container dom element have the fixed width and height. That's my idea. FYI.

**Comment by Farshadi73** [comment]
I have same problem in modal and specific tab .... 
I initialize editor after show that specific tab, by using jquery custom event difination

```
function onTabChange(element){
       ..... other code to handle show a specific tab
       element.trigger(tabname + '_active', tabname);
}

    // $editor defied globally
    $('body').on('your_tab_name_active', function () {

        if (typeof $editor === "undefined") {

            $editor = CodeMirror.fromTextArea(document.getElementById("landkit_code_code"), {
                mode: 'javascript',
                indentWithTabs: true,
                smartIndent: true,
                lineNumbers: true,
                lineWrapping: false,
                matchBrackets: true,
                autoRefresh: false,
                selectionPointer: true,
                styleActiveLine: true,
                direction: 'ltr',
                extraKeys: { "Ctrl-Space": "autocomplete" }
            });
        }
    });

```

**Comment by mennucc** [comment]
I have similar issue, I attach two proof-of-concepts, based on the trailingspace.html

[trailingspace_booster_tab.html.zip](https://github.com/codemirror/CodeMirror/files/7957560/trailingspace_booster_tab.html.zip)

in this first example I added Bootstrap and two tabs, each with the same content (up to changing `id`s). As noted above by  aishwaryardandi (on 30 Jul 2019), the second tab does not show the content at first, then after you click, the text is over the line numbers.

In the second example

[trailingspace_booster_tab_2.html.zip](https://github.com/codemirror/CodeMirror/files/7957574/trailingspace_booster_tab_2.html.zip)

similarly to what  Farshadi73 did ,  it triggers the CodeMirro code when the second tab is shown; the defect is that , each time you switch tab, a new editor pane is created; but as you see from the console , the code is run only once, so it is unclear where this behaviour is coming from.

To view the examples, unpack in the demo subdir.

Regards. a.
